### PR TITLE
[MaskRCNN] ONNX Dynamic Convolution Transpose

### PR DESCRIPTION
**Pull request by arogowie-intel**

### CODE REVIEW

File: src/ngraph/frontend/onnx_import/op/conv_transpose.cpp [hunk 1]
```diff
@@ -38,20 +42,237 @@ namespace ngraph
...
+                            const auto one_node =
+                                default_opset::Constant::create(element::i64, Shape{1}, {1});
+                            const auto two_node =
+                                default_opset::Constant::create(element::i64, Shape{1}, {2});
+                            const auto remaining_shape_length =
+                                std::make_shared<default_opset::Subtract>(conv_rank, two_node);
+                            const auto remaining_bias_shape_ones =
+                                std::make_shared<default_opset::Broadcast>(
+                                    one_node,
+                                    remaining_shape_length); // To skalar czy [1]?
```
Comment (tomdol) [inline src/ngraph/frontend/onnx_import/op/conv_transpose.cpp:208]: The comment :)

File: src/ngraph/frontend/onnx_import/utils/convpool.cpp [hunk 2]
```diff
@@ -38,43 +38,78 @@ namespace ngraph
...
+
             Strides get_dilations(const Node& node)
             {
                 return detail::get_attribute_value(node, "dilations");
             }
 
+            Strides get_dilations(const Node& node, const std::size_t kernel_rank)
+            {
+                return detail::get_attribute_value(node, "dilations", kernel_rank);
+            }
```
Comment (tsocha) [inline src/ngraph/frontend/onnx_import/utils/convpool.cpp:111]: as above
```suggestion
            Strides get_dilations(const Node& node, const std::size_t kernel_rank)
            {
                return detail::get_attribute_value(node, "dilations", kernel_rank);
            }
```

File: src/ngraph/frontend/onnx_import/utils/convpool.hpp [hunk 3]
```diff
@@ -33,19 +33,44 @@ namespace ngraph
...
-            /// \return The Strides object containing number of pixels for filter dilation
-            ///         (height, width, depth).
+            /// \brief      Get number of pixels to stride operation by in each direction.
+            ///
+            /// \param      node         The Node ptr representing Conv or Pool operation.
+            /// \param[in]  kernel_rank  The operator's kernel rank.
+            ///
+            /// \return     The kernel Shape object representing its dimensions (height, width,
+            ///             depth).
+            Strides get_strides(const Node& node, const std::size_t kernel_rank);
```
Comment (tsocha) [inline src/ngraph/frontend/onnx_import/utils/convpool.hpp:53]: These headers should be updated to accept a default value for kernel_rank.

File: src/ngraph/frontend/onnx_import/utils/convpool.hpp [hunk 4]
```diff
@@ -33,19 +33,44 @@ namespace ngraph
...
+            ///
+            /// \brief      Get number of pixels for filter dilation in each direction.
+            ///
+            /// \param      node         The Node ptr representing ONNX operation.
+            /// \param[in]  kernel_rank  The operator'skernel rank.
+            ///
+            /// \return     The Strides object containing number of pixels for filter dilation
+            ///             (height, width, depth).
+            Strides get_dilations(const Node& node, const std::size_t kernel_rank);
+
```
Comment (tsocha) [inline src/ngraph/frontend/onnx_import/utils/convpool.hpp:73]: As above
```suggestion
            ///
            /// \brief      Get number of pixels for filter dilation in each direction.
            ///
            /// \param      node         The Node ptr representing ONNX operation.
            /// \param[in]  kernel_rank  The operator'skernel rank.
            ///
            /// \return     The Strides object containing number of pixels for filter dilation
            ///             (height, width, depth).
            Strides get_dilations(const Node& node, const std::size_t kernel_rank = 0);

```
Please be consistent with the `[in]` part of the param.

File: src/ngraph/frontend/onnx_import/op/conv_transpose.cpp [hunk 5]
```diff
@@ -38,20 +42,237 @@ namespace ngraph
...
+                                data,
+                                filters,
+                                strides,
+                                pads_begin,
+                                pads_end,
+                                dilations,
+                                auto_pad_type,
+                                CoordinateDiff(std::begin(output_padding),
+                                               std::end(output_padding)));
+                        }
```
[the remark this answers is not in the document]
Comment (arogowie-intel) [reply src/ngraph/frontend/onnx_import/op/conv_transpose.cpp:83]: Good point

File: src/ngraph/frontend/onnx_import/op/conv_transpose.cpp [hunk 6]
```diff
@@ -38,20 +42,237 @@ namespace ngraph
...
+                                         "\"kernel_shape\" attribute is required if input data rank"
+                                         " is dynamic.");
+                        std::vector<std::size_t> kernel_shape =
+                            node.get_attribute_value<std::vector<std::size_t>>("kernel_shape");
+
+                        strides = convpool::get_strides(node, kernel_shape.size());
+                        dilations = convpool::get_dilations(node, kernel_shape.size());
+                        paddings = convpool::get_pads(node, kernel_shape.size());
+                        num_spatial_dims = kernel_shape.size();
+                    }
```
[the remark this answers is not in the document]
Comment (arogowie-intel) [reply src/ngraph/frontend/onnx_import/op/conv_transpose.cpp:274]: I've added reading number of spatial dims from filter input.

File: src/ngraph/frontend/onnx_import/op/conv_transpose.cpp [hunk 7]
```diff
@@ -64,88 +285,46 @@ namespace ngraph
 
                     int64_t groups{node.get_attribute_value<int64_t>("group", 1)};
 
-                    CHECK_VALID_NODE(node,
-                                     ((groups >= 0) &&
-                                      (groups <= static_cast<int64_t>(data->get_shape().at(1))) &&
-                                      (groups <= static_cast<int64_t>(filters->get_shape().at(0)))),
-                                     "incorrect value of 'group' attribute: ",
-                                     groups);
```
[the remark this answers is not in the document]
Comment (arogowie-intel) [reply src/ngraph/frontend/onnx_import/op/conv_transpose.cpp:72]: Didn't want to duplicate checks. Correctness of this value would be checked during specific convolution operation shape inference .

**Pull request merged**